monai/auto3dseg/analyzer.py

```diff
@@ -0,0 +1,688 @@
+
+
+AnalyzerType = TypeVar("AnalyzerType", bound=Analyzer)
+
+
+class ImageStats(Analyzer):
+    """
+    Analyzer to extract image stats properties for each case(image).
+
+    Args:
+        image_key: the key to find image data in the callable function input (data)
+        meta_key_postfix: the postfix to append for meta_dict ("image_meta_dict")
+
+    Examples:
+
+    .. code-block:: python
+
+        import numpy as np
+        from monai.auto3dseg.analyzer import ImageStats
+
+        input = {}
+        input['image'] = np.random.rand(1,30,30,30)
+        input['image_meta_dict'] = {'affine': np.eye(4)}
+        analyzer = ImageStats(image_key="image")
+        print(analyzer(input))
+
+    """
+
+    def __init__(
+        self, image_key: str, stats_name: str = "image_stats", meta_key_postfix: Optional[str] = "meta_dict"
+    ) -> None:
+
+        if not isinstance(image_key, str):
+            raise ValueError("image_key input must be str")
+
+        self.image_key = image_key
+        self.image_meta_key = f"{self.image_key}_{meta_key_postfix}"
+
+        report_format = {
+            str(ImageStatsKeys.SHAPE): None,
+            str(ImageStatsKeys.CHANNELS): None,
+            str(ImageStatsKeys.CROPPED_SHAPE): None,
+            str(ImageStatsKeys.SPACING): None,
+            str(ImageStatsKeys.INTENSITY): None,
+        }
+
+        super().__init__(stats_name, report_format)
+        self.update_ops(str(ImageStatsKeys.INTENSITY), SampleOperations())
+
+    def __call__(self, data):
+        """
+        Callable to execute the pre-defined functions
+
+        Returns:
+            A dictionary. The dict has the key in self.report_format. The value of
+            ImageStatsKeys.INTENSITY is in a list format. Each element of the value list
+            has stats pre-defined by SampleOperations (max, min, ....)
+
+        Note:
+            The stats operation uses numpy and torch to compute max, min, and other
+            functions. If the input has nan/inf, the stats results will be nan/inf.
+
+        """
+        d = dict(data)
+        start = time.time()
+        ndas = data[self.image_key]
+        ndas = [ndas[i] for i in range(ndas.shape[0])]
+        if "nda_croppeds" not in d:
+            nda_croppeds = [get_foreground_image(nda) for nda in ndas]
+
+        # perform calculation
+        report = deepcopy(self.get_report_format())
+
+        report[str(ImageStatsKeys.SHAPE)] = [list(nda.shape) for nda in ndas]
+        report[str(ImageStatsKeys.CHANNELS)] = len(ndas)
+        report[str(ImageStatsKeys.CROPPED_SHAPE)] = [list(nda_c.shape) for nda_c in nda_croppeds]
+        report[str(ImageStatsKeys.SPACING)] = np.tile(
+            np.diag(data[self.image_meta_key]["affine"])[:3], [len(ndas), 1]
+        ).tolist()
+        report[str(ImageStatsKeys.INTENSITY)] = [
+            self.ops[str(ImageStatsKeys.INTENSITY)].evaluate(nda_c) for nda_c in nda_croppeds
```

Comment:
How can we ensure the keys used here are consistent with the report format?
I feel we need to add some check at least, for example, the key must exist in the report format.

Thanks.

Reply:
Agree. I added a report verification in the end of the these calls.